Utilities/bootstrap

```diff
@@ -261,6 +269,12 @@ class Target(object):
             import_paths.append(os.path.join(args.libdispatch_build_dir, "src", "swift"))
         if args.libdispatch_source_dir:
             import_paths.append(args.libdispatch_source_dir)
+            if platform.system() == 'Darwin':
+                other_args.extend(["-Xcc", "-DDEPLOYMENT_TARGET_MACOSX"])
```

Comment:
@parkera: aren't these macros kind of discouraged?

Reply:
We absolutely need the `DEPLOYMENT_TARGET_*` macros, because the public headers include headers like `fts.h` which is a BSD extension.